WordPress/src/test/java/org/wordpress/android/ui/posts/editor/StorePostViewModelTest.kt

```diff
@@ -229,4 +232,26 @@ class StorePostViewModelTest : BaseUnitTest() {
         verify(postUtils).trackSavePostAnalytics(postModel, site)
         verify(uploadService).uploadPost(context, postId, isFirstTimePublish)
     }
+
+    @Test
+    fun `updates progress dialog LiveData appropriately`() {
+        val actual = mutableListOf<DialogVisibility>()
+        val expected = mutableListOf<DialogVisibility>()
+        viewModel.savingInProgressDialogVisibility.observeForever {
+            actual.add(it)
+        }
+
+        // is Hidden upon initialization
+        expected.add(Hidden)
+        assertThat(actual).isEqualTo(expected)
+
+        viewModel.showSaveProgressDialog()
+        expected.add(Showing)
+        assertThat(actual).isEqualTo(expected)
+
+        val randomNonNullFinishState = SAVED_LOCALLY
+        viewModel.finish(randomNonNullFinishState)
+        expected.add(Hidden)
+        assertThat(actual).isEqualTo(expected)
+    }
```

Comment:
I was a bit conflicted about whether the ViewModel changes really needed unit tests since they're just basically passed along method calls (if method a gets called, make sure method b is called), but I do like to test view models, so I split the difference by writing this test so that it kind-of, sort-of tests the "lifecycle" of the progress dialog's live data. Happy to change this to a more standard set of unit tests if you prefer (i.e., separate tests for initialization, showing the dialog, and hiding the dialog).

Reply:
LGTM ;)